lib/msal-common/src/logger/Logger.ts

```diff
@@ -33,6 +33,141 @@ export interface ILoggerCallback {
     (level: LogLevel, message: string, containsPii: boolean): void;
 }
 
+/**
+ * Represents a single logged message with metadata
+ */
+export interface LoggedMessage {
+    hash: string;
+    level: LogLevel;
+    containsPii: boolean;
+    milliseconds: number;
+}
+
+/**
+ * LRU cache node for correlation ID management
+ */
+interface CorrelationLogData {
+    logs: LoggedMessage[];
+    firstEventTime: number;
+}
+
+// Shared cache state for better minification - using Map's insertion order for LRU
+const CACHE_CAPACITY = 10;
```

Comment:
This may be too small, given the number of tokens some apps request + initialize + getAccount calls, it can really add up in a short amount of time.

Reply:
What number is more reasonable from your pespective? I guess 50 should be enough